components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/dao/impl/UserSessionDAOImpl.java

```diff
@@ -89,4 +93,42 @@ public UserSession getSession(String sessionId) throws SessionManagementServerEx
         }
         return null;
     }
+
+    private void generateApplicationFromAppID(List<Application> applications) throws SessionManagementServerException {
+
+        List<String> appIDList = applications.stream()
+                .map(Application::getAppId)
+                .collect(Collectors.toList());
+        String placeholder = String.join(", ", Collections.nCopies(appIDList.size(), "?"));
+        String sql = SQLQueries.SQL_GET_APPLICATION.replace(SCOPE_LIST_PLACEHOLDER, placeholder);
+        try (Connection connection = IdentityDatabaseUtil.getDBConnection(false)) {
+            try (PreparedStatement ps = connection.prepareStatement(sql)) {
+                int index = 1;
+                for (String appId : appIDList) {
+                    ps.setString(index, appId);
+                    index++;
+                }
+                try (ResultSet rs = ps.executeQuery()) {
+                    while (rs.next()) {
+                        applications.get(rs.getRow() - 1).setAppName(rs.getString(1));
```

Comment:
can we guarantee that the result set and the applications are in the same order?